File: 611-valid-triangle-number/valid-triangle-number.ts
function triangleNumber(nums: number[]): number {
    // addition of 2 sides is always greater than 3rd
    nums = nums.sort((a, b)=> a - b)
    let countAns = 0;

    for(let i=nums.length-1; i>=0; i--){
        let left = 0;
        let right = i-1;

        while(left < right){
            if(nums[left] + nums[right] > nums[i]){
                countAns += right - left
                right--;
            }
            else{
                left++;
            }
        }
    }

    return countAns;
};

File: 611-valid-triangle-number/valid-triangle-number.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        //addition of two sides a+b is always greater than c
        sort(nums.begin(), nums.end());

        int count = 0;
        for(int c=nums.size()-1; c>=0; c--){
            int a = 0;
            int b = c-1;

            while(a < b){
                if(nums[a] + nums[b] > nums[c]){
                    count += b - a;
                    b--;
                }
                else{
                    a++;
                }
            }
        }

        return count;
    }
};